### PR TITLE
Extend SDK with functions for locking and unlocking/reloading checkout session

### LINKED ISSUE #2: Extend SDK with functions for locking and unlocking/reloading checkout session
Once support is implemented in the checkout frontend and checkout service extend SDK to allow merchants to 
- [ ] Lock checkout UI so the session can be updated in a backend-to-backend API request
- [ ] Signal to the checkout that the session has been updated and that the UI lock can be lifted when the session is reloaded.

Rel. https://github.com/Dintero/checkout-service/issues/984
Rel. https://github.com/Dintero/checkout/issues/571

### PR DESCRIPTION
**Pull request by andersem**
Closes #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,14 @@ _The checkout sdk will add a polyfill for promises if the browser does not suppo
                 console.log("href", event.href);
                 checkout.destroy();
             },
+            onSessionLocked: function(event, checkout) {
+                console.log("pay_lock_id", event.pay_lock_id);
+                checkout.destroy();
```
Comment (svnv) [inline README.md:86]: probably no need to destroy here?
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -81,6 +81,14 @@ _The checkout sdk will add a polyfill for promises if the browser does not suppo
...
+            },
+            onSessionLockFailed: function(event, checkout) {
+                console.log("session lock failed");
+                checkout.destroy();
```
Comment (svnv) [inline README.md:90]: probably no need to destroy here?

```suggestion
```

File: src/subscribe.ts [hunk 3]
```diff
@@ -40,6 +40,30 @@ const postAck = (iframe: HTMLIFrameElement, event: MessageEvent) => {
     }
 };
 
+/**
+ * Post a SessionLock-event to the checkout iframe.
+ */
+export const postSessionLock = (iframe: HTMLIFrameElement) => {
+    if (iframe.contentWindow) {
+        iframe.contentWindow.postMessage(
```
Comment (svnv) [inline src/subscribe.ts:48]: would it make sense to expand the post message bodies with a a sid so we can verify in the checkout that the sids match before we accept the message?

Comment (andersem) [reply src/subscribe.ts:48]: Added to both messages

File: README.md [hunk 4]
```diff
@@ -122,9 +128,53 @@ const checkout = await dintero.embed({
         console.log("href", event.href);
         checkout.destroy();
     },
+    onSessionLocked: (event, checkout) => {
+        console.log("pay_lock_id", event.pay_lock_id);
+        checkout.destroy();
+    },
+    onSessionLockFailed: (event, checkout) => {
+        console.log("session lock failed");
+        checkout.destroy();
```
Comment (svnv) [inline README.md:137]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -122,9 +128,53 @@ const checkout = await dintero.embed({
         console.log("href", event.href);
         checkout.destroy();
     },
+    onSessionLocked: (event, checkout) => {
+        console.log("pay_lock_id", event.pay_lock_id);
+        checkout.destroy();
```
Comment (svnv) [inline README.md:133]:
```suggestion
```

**Pull request merged**